core/auth/src/test/java/software/amazon/awssdk/auth/credentials/DefaultCredentialsProviderTest.java

```diff
@@ -97,6 +109,60 @@ void resolveCredentials_ProfileCredentialsProviderWithSupplierProfileFile_return
         });
     }
 
+    @Test
+    void resolveCredentials_DefaultCredentialProviderWithReloadWhenModified(@TempDir Path parentDirectory) throws Exception {
+        Path credentialsFilePath = generateTestCredentialsFile(parentDirectory, "customAccess", "customSecret");
+        SystemSettingUtilsTestBackdoor.addEnvironmentVariableOverride(ProfileFileSystemSetting.AWS_SHARED_CREDENTIALS_FILE.environmentVariable(),
+                                                                      credentialsFilePath.toString());
+        DefaultCredentialsProvider provider = DefaultCredentialsProvider.create();
+
+        assertThat(provider.resolveCredentials()).satisfies(awsCredentials -> {
+            assertThat(awsCredentials.accessKeyId()).isEqualTo("customAccess");
+            assertThat(awsCredentials.secretAccessKey()).isEqualTo("customSecret");
+        });
+
+        Path credentialsFilePath2 = generateTestCredentialsFile(parentDirectory,"modifiedAccess", "modifiedSecret");
+        assertThat(credentialsFilePath2).isEqualTo(credentialsFilePath);
+
+        int maxRetries = 4;
+        int i = 0;
+        // check if this can be replaced with RetryPolicy and backoff
```

Comment:
Can we provide a more meaningful comment here ? Like 

"Configuring test with a Retry and backoff to ensure credential loading takes effect"

Reply:
NIT: Could we also include a smal note about why we need to retry here? Something like "Waiting until file changes reach the SDK" or something like that?

Reply:
@anirudh9391 @L-Applin sure, will update the note